packages/flutter_test/lib/src/event_simulation.dart

```diff
@@ -85,24 +90,57 @@ class KeyEventSimulator {
         break;
     }
     for (final int code in map.keys) {
-      if (key.debugName == map[code].debugName) {
+      if (key.keyId == map[code].keyId) {
         keyCode = code;
         break;
       }
     }
     return keyCode;
   }
 
+  static PhysicalKeyboardKey _findPhysicalKey(LogicalKeyboardKey key, String platform) {
+    assert(_osIsSupported(platform), 'Platform $platform not supported for key simulation');
+    Map<int, PhysicalKeyboardKey> map;
+    switch (platform) {
```

Comment:
This code doesn't support windows directly, although the current implementation on Windows uses GLFW, which is the same as Linux.  Both of those are changing though.

The new key event design will address being able to fully synthesize key events so as to be indistinguishable from key events coming from the platform, and since it's no longer platform specific, will work for all platforms.